crescent/commands/signature.py

```diff
@@ -31,18 +31,18 @@
 
 
 def _unwrap_optional(origin: type[Any]) -> Any:
-    if get_origin(origin) is not Union:
-        return origin
-
     args = get_args(origin)
 
-    if len(args) != 2 or NoneType not in args:
-        return args
+    if len(args) == 2:
+        if args[0] is NoneType:
+            return args[1]
+        if args[1] is NoneType:
+            return args[0]
 
-    if args[1] is NoneType:
-        return args[0]
+    if get_origin(origin) is not Union:
```

Comment:
You already called `get_origin(origin)` and stored the result in args. This can just be `if args is not Union` I think.

Reply:
args is `args = get_args(origin)`

Reply:
nevermind